components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -239,8 +231,8 @@ public Response getAllAPIs(Integer limit, Integer offset, String xWSO2Tenant, St
                 RestApiUtil.handleMigrationSpecificPermissionViolations(tenantDomain, username);
             }*/
             Map<String, Object> result;
-
-            result = apiProvider.searchPaginatedAPIs(query, tenantDomain, offset, limit);
+            String orgId = PublisherCommonUtils.getOrgId(organizationId);
```

Comment:
are there multiple getOrgId methods?

Reply:
Yes, there are 2 different **orgId** retrieval methods.

1. Publisher - In Publisher tenant domain is not coming from the user as a request parameter. We are getting it from the context and pass it to the workflow. So we only need to pass **organizationId** to the method logic.

**_Method : PublisherCommonUtils.getOrgId(organizationId)_**

2. In Dev-Portal - Here tenantDomain is an optional request parameter from the user. Thus we need to incorporate that to create an **orgId**. So we need to pass **organizationId** as well as **xWSO2Tenant** to the method logic. If **xWSO2Tenant**  is null we are getting the tenantDomain from the context.

**_Method : RestApiCommonUtil.getOrgIdMatchesTenantDomain(organizationId, xWSO2Tenant)_**